src/languageSupport/languages/influxql/influxql.ts

```diff
@@ -0,0 +1,299 @@
+    'COUNT',
+    'DISTINCT',
+    'INTEGRAL',
+    'MEAN',
+    'MEDIAN',
+    'MODE',
+    'SPREAD',
+    'STDDEV',
+    'SUM',
+    // Selectors
+    'BOTTOM',
+    'FIRST',
+    'LAST',
+    'MAX',
+    'MIN',
+    'PERCENTILE',
+    'SAMPLE',
+    'TOP',
+    // Transformations
+    'ABS',
+    'ACOS',
+    'ASIN',
+    'ATAN',
+    'ATAN2',
+    'CEIL',
+    'COS',
+    'CUMULATIVE_SUM',
+    'DERIVATIVE',
+    'DIFFERENCE',
+    'ELAPSED',
+    'EXP',
+    'FLOOR',
+    'HISTOGRAM',
+    'LN',
+    'LOG',
+    'LOG2',
+    'LOG10',
+    'MOVING_AVERAGE',
+    'NON_NEGATIVE_DERIVATIVE',
+    'NON_NEGATIVE_DIFFERENCE',
+    'POW',
+    'ROUND',
+    'SIN',
+    'SQRT',
+    'TAN',
+    // Technical analysis
+    'HOLT_WINTERS',
+    'CHANDE_MOMENTUM_OSCILLATOR',
+    'EXPONENTIAL_MOVING_AVERAGE',
+    'DOUBLE_EXPONENTIAL_MOVING_AVERAGE',
+    'KAUFMANS_EFFICIENCY_RATIO',
+    'KAUFMANS_ADAPTIVE_MOVING_AVERAGE',
+    'TRIPLE_EXPONENTIAL_MOVING_AVERAGE',
+    'TRIPLE_EXPONENTIAL_DERIVATIVE',
+    'RELATIVE_STRENGTH_INDEX',
+  ],
+  tokenizer: {
+    root: [
+      // the order of the following operators in increasing precedence
+      {include: '@comments'},
+      {include: '@whitespace'},
+      [/([0-9]+(y|mo|w|d|h|m|s|ms|us|µs|ns))+/, 'literal.duration'],
+      {include: '@numbers'},
+      {include: '@strings'},
+      {include: '@complexIdentifiers'},
+      {include: '@delimiters'},
+      [/(TRUE|FALSE)/, 'literal.boolean'],
+      [
+        /[\w@#$]+/,
+        {
+          cases: {
+            '@keywords': 'keyword',
+            '@operators': 'operator',
+            '@illegal': 'illegal',
+            '@literals': 'predefined',
+            '@builtinFunctions': 'builtin.function',
+            '@default': 'identifier',
+          },
+        },
+      ],
+      [/[<>=!%&+\-*/|~^]/, 'operator'],
+    ],
+    comments: [
+      [/--+.*/, 'comment'],
+      [/\/\*/, {token: 'comment.quote', next: '@comment'}],
+    ],
+    comment: [
+      [/[^*/]+/, 'comment'],
+      // Not supporting nested comments, as nested comments seem to not be standard?
+      // i.e. http://stackoverflow.com/questions/728172/are-there-multiline-comment-delimiters-in-sql-that-are-vendor-agnostic
+      // [/\/\*/, { token: 'comment.quote', next: '@push' }],    // nested comment not allowed :-(
+      [/\*\//, {token: 'comment.quote', next: '@pop'}],
+      [/./, 'comment'],
+    ],
+    whitespace: [[/\s+/, 'white']],
+    numbers: [
+      [/0[xX][0-9a-fA-F]*/, 'literal.number'],
+      [/[$][+-]*\d*(\.\d*)?/, 'literal.number'],
+      [/((\d+(\.\d*)?)|(\.\d+))([eE][\-+]?\d+)?/, 'literal.number'],
+    ],
+    strings: [[/'/, {token: 'literal.string', next: '@string'}]],
+    string: [
+      [/[^']+/, 'literal.string'],
+      [/''/, 'literal.string'],
+      // support single quotes with backslash escape
+      // https://docs.influxdata.com/influxdb/v2.6/reference/syntax/influxql/spec/#strings
+      [/^'((?:\\.|[^\\'])*)'$/, 'literal.string'],
```

Comment:
This is not working properly:
`'str\'ing'`

Reply:
Done